Assets/Scripts/RangedWeapon.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RangedWeapon : MonoBehaviour
+{
+    public float speed;
+    protected EntityController archery;
+    protected EntityController target;
+    protected Vector3 startPosition;
+    protected Vector3 targetPosition;
+    protected int power;
+    protected bool isAlive = true;
+
+    public void Initialize(EntityController archer, EntityController targetEntity)
+    {
+        archery = archer;
+        target = targetEntity;
+        power = archer.unitData.damage;
+        targetPosition = targetEntity.transform.position;
+
+        Vector3 arrowSize = transform.lossyScale;
+
+        SpriteRenderer playerSpriteRenderer = archer.transform.Find("Sprite").GetComponent<SpriteRenderer>();
+        SpriteRenderer spriteRenderer = transform.Find("Sprite").GetComponent<SpriteRenderer>();
+
+        float playerWidth = playerSpriteRenderer.bounds.size.x;
+        float playerHeight = playerSpriteRenderer.bounds.size.y;
+
+        float fireballWidth = spriteRenderer.bounds.size.x;
+        float fireballHeight = spriteRenderer.bounds.size.y;
+
+        float offsetX = (playerWidth - fireballWidth) / 2f;
+        float offsetY = (playerHeight - fireballHeight) / 2f;
+
+        Vector3 fireballPosition = new Vector3(archer.transform.position.x + offsetX, archer.transform.position.y + offsetY, transform.position.z);
+
+        transform.position = fireballPosition;
+
+        Vector3 direction = target.transform.position - transform.position;
+        transform.rotation = Quaternion.LookRotation(Vector3.forward, direction);
+
+        float distance = Vector3.Distance(transform.position, target.transform.position);
+        float timeToReachTarget = distance / 5f;
+        speed = distance / timeToReachTarget;
+    }
+
+    protected void Start()
+    {
+        startPosition = transform.position;
+    }
+
+    protected void HittingEnemy()
+    {
+        if (archery is EnemyController)
+        {
+            foreach (UnitController unit in BattleManager.Instance.totalPlayerUnits)
+            {
+                SpriteRenderer unitSpriteRenderer = unit.transform.Find("Sprite").GetComponent<SpriteRenderer>();
+                SpriteRenderer rangedWeapon = transform.Find("Sprite").GetComponent<SpriteRenderer>();
+
+                if (unitSpriteRenderer.bounds.Intersects(rangedWeapon.bounds))
```

Comment:
This comment here is more of a warning than a suggestion of something to change, since we are starting to lack the time to fix every single detail.

That said, for hitbox collisions are the like, we should use trigger colliders instead of running such a method as this every frame, like you did here.
Notice how you are, every frame, iterating over every unit in the game, fetching each renderer, fetching the source's renderer and checking if a sprite intersection occurs.
With a trigger, you get notified when a projectile enters another collider. This reduces the computational cost drastically.

Again, not saying you have to fix this, we don't really have too much time in our hands anymore. But please keep this in mind from now on.